Assignment 2.Rmd

```diff
@@ -79,16 +93,109 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+names(tcount)<-c("id","COUNT")
+
+#number of disciplinary actions received by student
+scount<-group_by(EDGE, to) %>%
+  summarise(sum(count))
+names(scount)<-c("id","COUNT")
+
+#combine tcount with scount for disc actions per id
+discact<-bind_rows(tcount, scount)
+VERTEX <- left_join(VERTEX, discact, by = "id")
+VERTEX <- transform(VERTEX, group = as.factor(group))
+```
+```{r}
+#Graph
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot.igraph(g, vertex.size=VERTEX$COUNT*3,
+            edge.arrow.width=0,
+            vertex.color=VERTEX$group,
+            vertex.frame.color = NA,
+            edge.width=EDGE$count, 
+            layout=layout_nicely)
+```
+
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
+Creating a person-class matrix:
+```{r}
+#Data wrangling
+library(tidyr)
+D3<-read.csv("HUDK4050-2018-classes.csv")
+D3<-select(D3, num_range("Q", 1:9))
+D3<-D3[,c("Q8","Q9","Q1","Q3","Q4","Q5","Q6")]
+D3<-unite(D3,Full.name,Q8,Q9)
+D3 <- D3[-c(1,2,3,39), ]
+colnames(D3)<-c("Name", "C1", "C2", "C3", "C4", "C5")
+D3$Name <- gsub("_", " ", D3$Name)
+D3 <- gather(D3, "Order", "Class Name", 2:6)
+
+#Converts values in "Order" to 1
+D3[,2] <- 1
+
+#Removes rows that are blank or have HUDK 4050
+D3[D3==""] <- NA
+D3$`Class Name`[D3$`Class Name` == "HUDK 4050"] <- NA
+D3$`Class Name`[D3$`Class Name` == "HUDK4050"] <- NA
+D3 <- D3[complete.cases(D3), ]
+
+#Creates person-class data frame
+D4 <- spread(D3, key="Class Name",value="Order", fill = NA, convert = FALSE, drop = TRUE)
+row.names(D4) <- D4$Name
+D4 <- D4[,-1]
+
+#Converts NAs in D4 to 0
+D4[is.na(D4)] <- 0
+
+#Converts D4 to person-class matrix
+PCMatrix <- as.matrix(D4)
+```
+
+Creating person-person matrix:
+```{r}
+PCMatrix2 <- t(PCMatrix)
+PPMatrix <- PCMatrix %*% PCMatrix2
+diag(PPMatrix) <- NA
+```
+
+Graph of matrix:
+```{r}
+g2<-graph.adjacency(PPMatrix, weighted=T,mode="undirected")
+g2<-simplify(g2, remove.multiple = TRUE, remove.loops = TRUE)
+plot.igraph(g2, vertex.frame.color = NA, 
+            layout=layout_nicely)
+```
+
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+#Top 5 by Betweenness Centrality
+betw <- betweenness(g2)
+descbetw <- sort.int(bet,decreasing=TRUE,index.return=FALSE)
+descbetw[1:5]
+```
+Elliot is the most central in terms of betweenness (number of shortest paths that pass through node) followed by Xixuan, Sam, Juan, and Anna.
+
+```{r}
+#Top 5 by Degree Centrality
+deg <- degree(g2)
+descdeg <- sort.int(deg,decreasing=TRUE,index.return=FALSE)
+descdeg[1:5]
+```
```

Comment:
Can you interpret substantively what the difference is between betweenness and degree? Why are they different?